[#56] feat: 홈 → 글쓰기 화면 이동

jwonyLee opened the pull request:
## 개요

홈 → 글쓰기 화면 이동

## 시연

![글쓰기 화면 이동](https://user-images.githubusercontent.com/15073405/135744257-96f53cab-8229-40c7-8454-35f1ecd7953f.gif)

## 작업 사항

- `UIView`의 `parentViewController`를 찾는 메서드 구현
- `WriteViewController` 구현
  - `WriteViewModel.writeTypeSubject`에 따라 화면에 있는 텍스트 업데이트
- `WriteViewModel` 구현
- `TabBarVC` → `WriteVC` 이동 구현

### Linked Issue

close #56

jwonyLee committed:
[#56] feat: findParentViewController() in UIView+

jwonyLee committed:
[#56] feat: WriteViewController 생성

jwonyLee committed:
[#56] feat: WriteCoordinator 구현

jwonyLee committed:
[#56] feat: TabBarVC -> WriteVC 이동 구현

jwonyLee committed:
[#56] feat: WriteType에 따라 WriteVC 화면 구성 변

gustn3965 commented on ThingLog/Extension/UIView+.swift:
```diff
@@ -11,4 +11,16 @@ extension UIView {
+
+    /// 현재 뷰의 부모 뷰 컨트롤러를 찾는다.
+    /// - Returns: 현재 뷰의 부모 뷰 컨트롤러(optional)를 반환한다. 
+    func findParentViewController() -> UIViewController? {
```
이 메서드는 필요없을 것 같습니다. 이 메서드를 사용하는 곳은 결국 `PageViewController`를 찾기 위함인데, `PageViewController`에서 관리하면 필요가 없겠죵

gustn3965 commented on ThingLog/View/ChoiceWritingView.swift:
```diff
@@ -76,4 +89,17 @@ extension ChoiceWritingView {
+              }
+
+        parentViewController.writeTypeSubject.onNext(type)
+    }
```
위에서 `findParentViewController`가 필요없다고 했으니, 관련 코드는 필요가 없을 것 같구요. 
또, 직접적으로 `parentViewController`에 주입하는 것도 너무 강한 의존성이라 좋지 않아보입니다.
대신에 `selectedWriteTypeSubject`를 `ChocieWriitngView`에서 가지도록 하고, 이를 `PageViewController`에서 바인딩하는게 좋을 것 같습니다.

gustn3965 commented on ThingLog/ViewController/TabBarController.swift:
```diff
@@ -20,13 +28,18 @@ final class TabBarController: UITabBarController {
     }()
+
+    // MARK: - Properties
+    let writeTypeSubject: PublishSubject<WriteType> = PublishSubject<WriteType>()
```
이 프로퍼티는 없애고 `ChocieView`에서 갖도록 하는게 좋을 것 같습니다

gustn3965 commented on ThingLog/ViewController/TabBarController.swift:
```diff
@@ -86,6 +99,16 @@ final class TabBarController: UITabBarController {
     }
+
+    private func bindWriteType() {
+        writeTypeSubject
```
`chocieVew.selectedWriteTypeSubject`로 바인딩하면 좋을 것 같습니다.

gustn3965 reviewed (commented):
잘 구현하신 것 같습니다.
코멘트 남겨드렸으니 확인해주세요 ~

gustn3965 commented:
아, 그리구 뷰컨 전환하는 애니메이션은 딱히 병연님이 고려하지 않으셨어서 애니메이션은 false롣 두어도 괜찮을 것 같은데, 지원님 생각에 지금이 더 괜찮다면 여쭤보시는 것도 좋을 것 같아요!

jwonyLee committed:
[#56] fix: 피드백 반영
- writeTypeSubject -> selectedWriteTypeSubject
- selectedWriteTypeSubject 위치 변경

jwonyLee commented:
> 아, 그리구 뷰컨 전환하는 애니메이션은 딱히 병연님이 고려하지 않으셨어서 애니메이션은 false롣 두어도 괜찮을 것 같은데, 지원님 생각에 지금이 더 괜찮다면 여쭤보시는 것도 좋을 것 같아요!

애니메이션이 아예 없으면 부자연스럽게 화면 전환이 되서 일단은 있는 대로 진행할게요~

gustn3965 reviewed (approved):
LGTM 👍

jwonyLee committed:
Merge branch 'develop' into feature/56-WriteViewController
